src/com/griddynamics/yvoronovskyi/battleship/BattleshipField.java

```diff
@@ -0,0 +1,228 @@
+            if (!isShipCoordinateValid) {
+                System.out.println(Constants.COORDINATES_ERROR_MESSAGE);
+                continue;
+            }
+            Pattern pattern = Constants.COORDINATES_PATTERN;
+            Matcher matcher = pattern.matcher(coordinates);
+            matcher.matches();
+            String firstCoordinate = matcher.group(1);
+            String secondCoordinate = matcher.group(2);
+            Point pointA = getMapPoints(firstCoordinate);
+            Point pointB = getMapPoints(secondCoordinate);
+            isShipCoordinateValid = (pointA.getX() == pointB.getX() || pointA.getY() == pointB.getY())
+                    && !pointA.equals(pointB);
+            if (!isShipCoordinateValid) {
+                System.out.println(Constants.WRONG_SHIP_LOCATION_MESSAGE);
+
+            }
+            int sDesk = (pointA.x == pointB.x)
+                    ? Math.abs(pointA.y - pointB.y)
+                    : Math.abs(pointA.x - pointB.x);
+            isShipCoordinateValid = sDesk + 1 == desk;
+            if (!isShipCoordinateValid) {
+                System.out.println(Constants.WRONG_SHIP_LENGTH_MESSAGE);
+                continue;
+            }
+            isShipCoordinateValid = setShipOnMap(pointA, pointB, desk);
+            if (!isShipCoordinateValid) {
+                System.out.println(Constants.WRONG_TOO_CLOSE_LOCATION_MESSAGE);
+                continue;
+            }
+            System.out.println(this.printDesk(Constants.PRINT_ALL));
+        }
+    }
+
+    public boolean setShipOnMap(Point pointA, Point pointB, int desk) {
+        boolean isHorizontal = pointA.x == pointB.x;
+        int begin;
+        if (isHorizontal) {
+            begin = Math.min(pointA.y, pointB.y);
+            for (int index = begin; index < begin + desk; index++) {
+                if (!isOcupated(pointA.x, index)) {
+                    return false;
+                }
+            }
+            shipId++;
+            for (int index = begin; index < begin + desk; index++) {
+                makeOccupation(new Point(pointA.x, index), Constants.EMPTY_DELIMETER + shipId);
+            }
+        } else {
+            begin = Math.min(pointA.x, pointB.x);
+            for (int index = begin; index < begin + desk; index++) {
+                if (!isOcupated(index, pointA.y)) {
+                    return false;
+                }
+            }
+            shipId++;
+            for (int index = begin; index < begin + desk; index++) {
+                makeOccupation(new Point(index, pointA.y), Constants.EMPTY_DELIMETER + shipId);
+            }
+        }
+        return true;
+    }
+
+    private int getPrevious(int point) {
+        return (point == 0) ? point : point - 1;
+    }
+
+    private int getNext(int point) {
+        return (point == 9) ? point : point + 1;
+    }
+
+    private void makeOccupation(Point point, String shipId) {
```

Comment:
Please, consider not using Point object here

Reply:
Done